diff -Naur folly-2021.07.20.01.orig/folly/container/detail/F14Table.h folly-2021.07.20.01/folly/container/detail/F14Table.h
--- folly-2021.07.20.01.orig/folly/container/detail/F14Table.h	2021-07-22 14:11:34.000000000 -0700
+++ folly-2021.07.20.01/folly/container/detail/F14Table.h	2021-07-26 12:24:40.216884520 -0700
@@ -71,7 +71,6 @@
 #include <arm_neon.h> // uint8x16t intrinsics
 #else // SSE2
 #include <emmintrin.h> // _mm_set1_epi8
-#include <immintrin.h> // __m128i intrinsics
 #include <xmmintrin.h> // _mm_prefetch
 #endif
 
