--- folly-2022.03.14.00/folly/portability/Constexpr.h.gcc	2022-03-13 13:18:37.000000000 +0900
+++ folly-2022.03.14.00/folly/portability/Constexpr.h	2022-07-16 14:07:53.162390109 +0900
@@ -98,7 +98,11 @@
 
 template <typename Char>
 constexpr std::size_t constexpr_strlen(const Char* s) noexcept {
+#if __GNUC_PREREQ(11, 0)
+  return detail::constexpr_strlen_internal(s, 0u);
+#else
   return detail::constexpr_strlen_internal(s, 0);
+#endif
 }
 
 template <typename Char>
