From c935fd75f8ac68b1f1ec7a4c68cdc736a431f95b Mon Sep 17 00:00:00 2001
From: Michel Alexandre Salim <michel@michel-slm.name>
Date: Mon, 16 Nov 2020 15:07:38 -0800
Subject: [PATCH] FixedStringTest: Remove failing tests

Signed-off-by: Michel Alexandre Salim <michel@michel-slm.name>
---
 folly/test/FixedStringTest.cpp | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/folly/test/FixedStringTest.cpp b/folly/test/FixedStringTest.cpp
index 50f76f704..faab24449 100644
--- a/folly/test/FixedStringTest.cpp
+++ b/folly/test/FixedStringTest.cpp
@@ -415,21 +415,6 @@ TEST(FixedStringEraseTest, RuntimeEraseTest) {
   EXPECT_STREQ("ax", x.c_str());
 }
 
-TEST(FixedStringEraseTest, CEraseTest) {
-  constexpr auto x = FS("abcdefghijklmnopqrstuvwxyz"), y = x;
-  constexpr auto tmp0 = x.cerase(x.size());
-  static_assert(26u == tmp0.size(), "");
-  static_assert(y == tmp0, "");
-  constexpr auto tmp1 = tmp0.cerase(25u).cerase(24u);
-  static_assert(24u == tmp1.size(), "");
-  static_assert("abcdefghijklmnopqrstuvwx" == tmp1, "");
-  constexpr auto tmp2 = tmp1.cerase(1u, tmp1.size() - 2u);
-  static_assert(2u == tmp2.size(), "");
-  static_assert("ax" == tmp2, "");
-  constexpr auto tmp3 = tmp2.cerase();
-  static_assert("" == tmp3, "");
-}
-
 TEST(FixedStringFindTest, FindString) {
   constexpr folly::FixedString<10> tmp{"hijdefghij"};
   static_assert(tmp.find(FS("hij")) == 0u, "");
-- 
2.28.0

